server/controllers/shoppingList.js

```diff
@@ -205,24 +158,24 @@ const getListData = (req, resp) => {
         { purchaserIds: req.user._id }
       ]
     },
-    (err, documents) => {
+    (err, doc) => {
       if (err) {
-        return resp.status(404).send({ message: err.message });
+        return resp.status(400).send({
+          message:
+            'An error occurred while fetching the lists data. Please try again.'
+        });
       }
 
-      if (documents && documents.length > 0) {
-        const { cohortId, _id, isArchived } = documents[0];
-
-        if (isArchived) {
-          return resp.status(200).json({ cohortId, _id, isArchived });
-        }
+      if (doc) {
+        const list = _head(doc);
+        const { cohortId, _id, isArchived } = list;
 
-        return resp.status(200).json(documents[0]);
+        return isArchived
+          ? resp.status(200).json({ cohortId, _id, isArchived })
+          : resp.status(200).json(list);
       }
 
-      return resp
-        .status(404)
-        .send({ message: 'Products not found for given list id!' });
+      resp.status(404).send({ message: 'List data not found.' });
```

Comment:
I think that we should check if doc exist first and use quick return if not. There will be more logic when we add checking roles, so all code for creating success response should be at the end...